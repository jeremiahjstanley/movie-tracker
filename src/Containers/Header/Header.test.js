import React from 'react';
import { Header } from './index.js';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper;
  let mockLogOut;

  beforeEach(() => {
    mockLogOut = jest.fn();
    wrapper = shallow(<Header users={{ id: 7 }} favorites={['Happy birthday Casey Poe']} logOut={ mockLogOut } />)
  });

  it('should match the snapshot when there is a user', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there is no user', () => {
    const mockUser = {};
    wrapper = shallow(<Header users={mockUser} favorites={[]} logOut={mockLogOut} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call logOut when logOut button is clicked', () => {
    wrapper.find('.log-out').simulate('click');
    expect(mockLogOut).toHaveBeenCalled();
  });
});
