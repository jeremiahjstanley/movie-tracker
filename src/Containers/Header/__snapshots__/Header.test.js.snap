// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    favorites={
        Array [
            "Happy birthday Casey Poe",
          ]
    }
    logOut={[Function]}
    users={
        Object {
            "id": 7,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Cage Tracker Logo"
          className="logo"
          src="logo.png"
/>,
        <ul
          className="navigation-links"
>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="home-link"
                              to="/"
                    >
                              Home
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="login-link"
                              to="/login"
                    >
                              <a
                                        className="log-out"
                                        onClick={[Function]}
                              >
                                        Logout
                              </a>
                    </NavLink>
          </li>
          <li>
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="favorites-link"
                              to="/favorites"
                    >
                              Favorites
                    </NavLink>
          </li>
</ul>,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Cage Tracker Logo",
          "className": "logo",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="home-link"
                            to="/"
              >
                            Home
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="login-link"
                            to="/login"
              >
                            <a
                                          className="log-out"
                                          onClick={[Function]}
                            >
                                          Logout
                            </a>
              </NavLink>
</li>,
            <li>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="favorites-link"
                            to="/favorites"
              >
                            Favorites
              </NavLink>
</li>,
          ],
          "className": "navigation-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="home-link"
                to="/"
>
                Home
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "className": "home-link",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="login-link"
                to="/login"
>
                <a
                                className="log-out"
                                onClick={[Function]}
                >
                                Logout
                </a>
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": <a
                  className="log-out"
                  onClick={[Function]}
>
                  Logout
</a>,
                "className": "login-link",
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Logout",
                  "className": "log-out",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Logout",
                "type": "a",
              },
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="favorites-link"
                to="/favorites"
>
                Favorites
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Favorites",
                "className": "favorites-link",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Cage Tracker Logo"
            className="logo"
            src="logo.png"
/>,
          <ul
            className="navigation-links"
>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="home-link"
                                    to="/"
                        >
                                    Home
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="login-link"
                                    to="/login"
                        >
                                    <a
                                                className="log-out"
                                                onClick={[Function]}
                                    >
                                                Logout
                                    </a>
                        </NavLink>
            </li>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="favorites-link"
                                    to="/favorites"
                        >
                                    Favorites
                        </NavLink>
            </li>
</ul>,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Cage Tracker Logo",
            "className": "logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="home-link"
                                to="/"
                >
                                Home
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="login-link"
                                to="/login"
                >
                                <a
                                                className="log-out"
                                                onClick={[Function]}
                                >
                                                Logout
                                </a>
                </NavLink>
</li>,
              <li>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="favorites-link"
                                to="/favorites"
                >
                                Favorites
                </NavLink>
</li>,
            ],
            "className": "navigation-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="home-link"
                  to="/"
>
                  Home
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Home",
                  "className": "home-link",
                  "to": "/",
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="login-link"
                  to="/login"
>
                  <a
                                    className="log-out"
                                    onClick={[Function]}
                  >
                                    Logout
                  </a>
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": <a
                    className="log-out"
                    onClick={[Function]}
>
                    Logout
</a>,
                  "className": "login-link",
                  "to": "/login",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Logout",
                    "className": "log-out",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Logout",
                  "type": "a",
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="favorites-link"
                  to="/favorites"
>
                  Favorites
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Favorites",
                  "className": "favorites-link",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
